File: C++_preparation/BOJ/BOJ_dfs_bfs.cpp
#include <iostream>
#include <queue>
#include <memory.h>

using namespace std;

int m, n, v;
int dx[4] = {0, 0, -1, 1};
int dy[4] = {-1, 1, 0, 0};

int graph[1001][1001];
bool visited[1001][1001];

void dfs(int i, int j)
{
}
int main()
{
    cin >> m >> n >> v;
    memset(graph, 0, sizeof(graph));
    memset(visited, false, sizeof(visited));
    for (int i = 0; i < n; i++)
    {
        cin >> a >> b;
        graph[a][b] = 1;
        graph[b][a] = 1;
    }
    return 0;
}

File: C++_preparation/BOJ/BOJ_dfs_bfs.py
import heapq
from collections import deque

n, m, v = map(int, input().split())

graph = [[] for _ in range(n+1)]
visited = [False for _ in range(n+1)]


def dfs(start):
    visited[start] = True
    print(start, end=" ")
    for i in graph[start]:
        if not visited[i]:
            dfs(i)


def bfs(start):
    q = deque([])
    visited[start] = True
    q.append(start)
    while q:
        x = q.pop()
        print(x, end=" ")
        for i in graph[x]:
            if not visited[i]:
                q.appendleft(i)
                visited[i] = True


for i in range(m):
    a, b = map(int, input().split())
    graph[a].append(b)
    graph[b].append(a)

for i in range(n):
    graph[i+1].sort()
dfs(v)
print()
visited = [False for _ in range(n+1)]
bfs(v)
print()
